### PR TITLE
For #28201 - Make a LightDarkPreview annotation

### LINKED ISSUE #28201: Make a “@LightDarkPreview()” annotation that makes two previews from one annotation
Currently we use two annotations in Compose Preview functions in order to display previews with both dark mode on and off.

This task is for creating a multi-annotation class that wraps both of these annotations into a single annotation, helping with readability and any future extension of the multi-annotation so it will include other rendering options.

Tasks:
- [ ] Create multi-annotation for both dark mode on and off
- [ ] Replace all uses of the two annotations with the single multi-annotation

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-22148)

### PR DESCRIPTION
**Pull request by HarrisonOg**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #28201

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package org.mozilla.fenix.compose
+
+import android.content.res.Configuration
+import androidx.compose.ui.tooling.preview.Preview
+
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_YES)
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_NO)
+annotation class LightDarkPreview
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt:8]: let's move this initial change into its own commit, so we'll ultimately land two commits with this PR

File: app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package org.mozilla.fenix.compose
```
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt:1]: this file needs the license

Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt:1]: to prevent the root `.../compose/` directory from becoming too bloated, let's move this to `.../compose/annotation`. We'll add any more similar annotation classes there in the future.

**Review by MozillaNoah — COMMENTED** [review]
This looks great! I've left some comments for a few minor things before we land this. Just make sure to run detekt/lint on your branches before opening PRs or pushing new commits to help save some review cycles and get the CI all-green.

**Comment by gabrielluong** [comment]
Noice 👌

File: app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt (hunk 1 above)
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt:8]: 🚗 I'm not super familiar with annotation classes, but I assume `LightDarkPreview` doesn't expose the parameters available to `@Preview` by some transitive magic or anything. Should we consider adding those same parameters here? Is there a way to pass them to both of the super classes? 

I'm just concerned that we could lose some flexibility in the our preview options if this becomes a common usage.

File: app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt (hunk 2 above)
Comment (HarrisonOg) [reply app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt:1]: sounds good, will make the new directory for annotation

Comment (HarrisonOg) [reply app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt:1]: ah I knew this file was looking a little empty. fixing

File: app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt (hunk 1 above)
Comment (HarrisonOg) [reply app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt:8]: this is a good point, I'll look into the options for `@Preview`

**Comment by gabrielluong** [comment]
Just a reminder that we'll want to do a squash landing. It would also be helpful to start incorporating an amend/squash workflow since we want to avoid fixup-type commits landing in the tree (eg, "[fix import ordering](https://github.com/mozilla-mobile/fenix/pull/28202/commits/4254f344a7739a4651a733abe43b9de854f9c43f)").

File: app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt (hunk 1 above)
Comment (HarrisonOg) [reply app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt]: hi @MatthewTighe 

after some further investigation, I have discovered that we mainly use only three fields for the following scenarios:
1) locale : this is used for Right to Left and Left to Right display previews. 
2) uiMode: this is used for Night Mode on and off 
3) name: used for labeling  

This new multi-annotation preview class would mainly be used to replace option 2 as it is the most common, used just with the night mode on and off.

  I also discovered that I cannot pass the parameters up to the multi-annotations like in this example: 
```
@Preview(name = name, uiMode = Configuration.UI_MODE_NIGHT_YES)
@Preview(name = name, uiMode = Configuration.UI_MODE_NIGHT_NO)
annotation class LightDarkPreview(
    val name: String = "",
    val group: String = "",
    @IntRange(from = 1) val apiLevel: Int = -1,
    // TODO(mount): Make this Dp when they are inline classes
    val widthDp: Int = -1,
    // TODO(mount): Make this Dp when they are inline classes
    val heightDp: Int = -1,
    val locale: String = "",
    @FloatRange(from = 0.01) val fontScale: Float = 1f,
    val showSystemUi: Boolean = false,
    val showBackground: Boolean = false,
    val backgroundColor: Long = 0,
    @UiMode val uiMode: Int = 0,
    @Device val device: String = Devices.DEFAULT
)
```
 my proposal is that when more annotation fields are needed, use the original @Preview() annotation with the fields filled in while keeping this multi-annotation preview class for just the Night Mode on and off scenario.

**Review by MozillaNoah — APPROVED** [review]
LGTM. Just make sure to edit the message of the second commit to add the issue number

File: app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt (hunk 1 above)
Comment (MatthewTighe) [reply app/src/main/java/org/mozilla/fenix/compose/LightDarkPreview.kt]: That sounds good to me! Shame the parameters can't be shared. Seems like it would be super helpful to have the name propagate, for instance.
